_overviews/tutorials/binary-compatibility-for-library-authors.md

```diff
@@ -173,6 +173,54 @@ You can find detailed explanations, runnable examples and tips to maintain binar
 
 Again, we recommend using MiMa to double-check that you have not broken binary compatibility after making changes.
 
+### Changing case class definition in a compatible manner
+
+Sometimes it is desirable to be able to change the definition of a case class (adding and/or removing fields) while still staying compatible with the users of the case class, i.e. not breaking the so called _binary compatibility_.
+
+To achieve that, follow this pattern:
+ * make the constructor private
+ * define a private `unapply` function in the companion object (note that by doing that the case class loses the ability to be used in a pattern match)
```

Comment:
Well usage in patterns is possible e.g. type patterns `case p: Person =>` so it is "only" constructor patterns that are not possible if I'm correct? So `case Person(n,a) => ` would not work as `unapply` is private, but other patterns work. So I think this should be clarified as the reader might not know about all the unapply mechanics under the hood...

Reply:
Edit: I guess extractor patterns could be made to work if the library author provides custom extractors?

Reply:
Anyway, I think it would be good to explain the implications on pattern matching in more detail...